upgrades/4.1_5.0/README.md

```diff
@@ -0,0 +1,109 @@
+# TAMS 4.1 to 5.0 Upgrade
+
+**⚠️ IMPORTANT: This upgrade process is ONLY required when upgrading from version 4.1 to 5.0. Fresh installations of version 5.0 do NOT need this upgrade.**
+
+## Overview
+
+This CloudFormation template deploys a Lambda function that performs the necessary data transformations when upgrading from TAMS v4.1 to v5.0. The upgrade includes:
+
+- **Webhooks ETL**: Migrates WebHook data from DynamoDB to Neptune.
+- **Storage ETL**: Migrates FlowStorage data from the old table structure (with flow_id sort key) to the new structure (without sort key) and changes the `storage_ids` (list) field to `storage_id` (str) single value.
+- **Tags ETL**: Updates tag properties in Neptune to serialise the existing strings. V5.0 stores all tags as serialised JSON strings so existing values need serialising.
+
+## Prerequisites
+
+**⚠️ DOWNTIME REQUIRED: The TAMS system will be in an unusable state after upgrading the CloudFormation stack to v5.0 until all ETL processes in this Lambda function are completed. Plan for scheduled downtime during this upgrade.**
```

Comment:
Could this give an indication of the amount of time needed? I realise it'll depend on the amount of data to migrate, but "seconds" versus "minutes" versus "tens of minutes" versus "hours" could still be a useful indicator.